include/pybind11/numpy.h

```diff
@@ -319,6 +345,11 @@ class array : public buffer {
         return (size_t) PyArray_GET_(m_ptr, nd);
     }
 
+    /// Base object
+    object base() const {
```

Comment:
NumPy has a .base flag, so I thought it would be nice to expose it in the C++ API as well.

This feature (pointer into memory owned by a "base") is really neat for exposing a NumPy "view" into a data structure that may be implemented in a completely different programming language. The base pointer lets you get back from the numpy view to the parent class, which can be useful in a function that is called without knowing the latter.